language/bytecode-verifier/src/nonce.rs

```diff
@@ -5,7 +5,7 @@
 //! A Nonce instance represents an arbitrary reference or access path.
 //! The integer inside a Nonce is meaningless; only equality and borrow relationships are
 //! meaningful.
-#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, Ord, PartialOrd)]
+#[derive(Clone, Copy, Debug, Default, Hash, PartialEq, Eq, Ord, PartialOrd)]
```

Comment:
Cloning and copying a nonce? Hmm.

Reply:
Happy to change the type name Nonce.  In the type checker, a nonce simply represents an abstract identifier for a reference that is present somewhere in the locals or on the evaluation stack.  The invariant being maintained is that distinct locals and locations in the evaluation stack contain distinct nonces.  That is why the analysis needs a supply of fresh nonces.